pkg/services/sqlstore/dashboard.go

```diff
@@ -132,13 +132,34 @@ func SearchDashboards(query *search.FindPersistedDashboardsQuery) error {
 	var sql bytes.Buffer
 	params := make([]interface{}, 0)
 
-	sql.WriteString(`SELECT
-					  dashboard.id,
-					  dashboard.title,
-					  dashboard.slug,
-					  dashboard_tag.term
-					FROM dashboard
-					LEFT OUTER JOIN dashboard_tag on dashboard_tag.dashboard_id = dashboard.id`)
+	if len(query.Tags) > 0 {
+		sql.WriteString(`SELECT dashboard.id, dashboard.title, dashboard.slug, dashboard_tag.term
+							FROM
+							( SELECT dashboard.id, dashboard.title, dashboard.slug, dashboard_tag.term, dashboard.org_id
+							FROM dashboard
+							INNER JOIN dashboard_tag
+							ON dashboard_tag.dashboard_id = dashboard.id
+							WHERE dashboard_tag.term IN (`)
```

Comment:
the AND is done doing an OR and counting the occurences (see line 150) 
ROUP BY dashboard.id HAVING COUNT(dashboard.slug) >= ?)